com.unity.ml-agents/Runtime/Policies/BehaviorParameters.cs

```diff
@@ -232,6 +241,21 @@ internal IPolicy GeneratePolicy(ActionSpec actionSpec, ActuatorManager actuatorM
             }
         }
 
+        /// <summary>
+        /// Query the behavior parameters in order to see if the Agent is running in Heuristic Mode.
+        /// </summary>
+        /// <returns>true if the Agent is running in Heuristic mode.</returns>
+        public bool IsInHeuristicMode()
+        {
+            if (BehaviorType == BehaviorType.HeuristicOnly)
+            {
+                return true;
+            }
+            return BehaviorType == BehaviorType.Default &&
+                ReferenceEquals(Model, null) &&
+                !Academy.Instance.IsCommunicatorOn;
```

Comment:
Can you check `Academy.IsInitialized` before `Academy.Instance.IsCommunicatorOn` so that we don't accidentally instantiate the Academy before it's ready?